pkg/sql/executor.go

```diff
@@ -793,13 +819,12 @@ func (e *Executor) execParsed(
 		// This is where the magic happens - we ask db to run a KV txn and possibly retry it.
 		txn := txnState.mu.txn // this might be nil if the txn was already aborted.
 		err := txn.Exec(session.Ctx(), execOpt, txnClosure)
-		if err != nil && len(results) > 0 {
```

Comment:
Can I simply return an error here? Or does it _need_ to be attached to the last result. I'm not sure.